### PR TITLE
CashFusion: Show disabled icon and add message on non-fusable wallets

### PR DESCRIPTION
**Pull request by EchterAgo**
Just not showing the CashFusion icon on non-fusable wallets is bad UX because the user will never know why the button is missing. Instead we show the icon as disabled with a message indicating why it is disabled.

### CODE REVIEW

**Comment by cculianu** [comment]
This looks good!  I will finish reviewing it later but I am glad you put it in the `self.widgets` data structure which is responsible for cleanup on plugin unload.  Reading the code over once it looks good.  I will do a real review later.

File: plugins/fusion/qt.py [hunk 1]
```diff
@@ -524,6 +530,20 @@ def run(self):
...
+        super().__init__(icon_fusion_logo_gray, 'Fusion', self.show_message)
+
+        self.wallet = wallet
+        self.message = message
+
+        self.setToolTip(_("CashFusion ({disabled_message})").format(
+            disabled_message=message))
+
+    def show_message(self):
+        QMessageBox.critical(Plugin.get_suitable_dialog_window_parent(self.wallet),
```
Comment (cculianu) [inline plugins/fusion/qt.py:544]: 1. I'm glad you used that `get_suitable_dialoig_window_parent` here 👍 
2. I wonder if a less .. *alarming* `QMessageBox` is warranted here.  On macOS at least this dialog box looks a little bit scary (big red stop sign)... maybe `QMessageBox.information`?

File: plugins/fusion/qt.py [hunk 2]
```diff
@@ -163,22 +163,28 @@ def on_new_window(self, window):
...
-
-        want_autofuse = Conf(wallet).autofuse
-        self.add_wallet(wallet, window.gui_object.get_cached_password(wallet))
+        can_fuse = can_fuse_from(wallet) and can_fuse_to(wallet)
+        if can_fuse:
+            sbbtn = FusionButton(self, wallet)
+            self.server_status_changed_signal.connect(sbbtn.update_server_error)
+        else:
+            # If we can not fuse we create a dummy fusion button that just displays a message
+            sbbtn = DisabledFusionButton(wallet, _('disabled because this wallet type is not supported'))
```
Comment (cculianu) [inline plugins/fusion/qt.py:172]: On macOS the title doesn't always get shown in windows (stupid macos).  Also I wonder if the message can be made a little more "informative" and "helpful".. 

Something like...

`_('This wallet type ({wtype}) is cannot be used with CashFusion.\n\nPlease use a standard deterministic spending wallet with CashFusion.').format(wtype=wallet.wallet_type`

?

File: plugins/fusion/qt.py (hunk 1 above)
Comment (EchterAgo) [reply plugins/fusion/qt.py]: updated

File: plugins/fusion/qt.py (hunk 2 above)
Comment (EchterAgo) [reply plugins/fusion/qt.py]: updated

**Comment by EchterAgo** [comment]
I also fixed the issue where the icon wasn't in the correct state on startup.

**Comment by cculianu** [comment]
I tested it.. I reviewed it.  It looks good. Again, thanks Axel!

**Pull request merged**